src/main/cpp/src/ftos_converter.cuh

```diff
@@ -493,7 +493,7 @@ __device__ inline floating_decimal_64 d2d(uint64_t const ieeeMantissa, uint32_t
   uint64_t m2;
   if (ieeeExponent == 0) {
     // We subtract 2 so that the bounds computation has 2 additional bits.
-    e2 = 1 - DOUBLE_BIAS - DOUBLE_MANTISSA_BITS - 2;
+    e2 = static_cast<int32_t>(1 - DOUBLE_BIAS - DOUBLE_MANTISSA_BITS - 2);
```

Comment:
Please also modify line 499 below, using `static_cast` instead of C-style cast.

Reply:
OMG, there are too many instances of such C-style cast :(

Reply:
Modified every C-style cast I found